src/nvme/mi.c

```diff
@@ -580,6 +603,36 @@ static int nvme_mi_admin_parse_status(struct nvme_mi_resp *resp, __u32 *result)
 	return nvme_status;
 }
 
+static int nvme_mi_service_parse_status(struct nvme_mi_resp *resp, __le16 *cpsr)
+{
+	struct nvme_mi_service_resp_hdr *resp_hdr;
+
+	/* we have a few different sources of "result" here: the status header
+	 * in the MI response, the cdw3 status field, and (command specific)
+	 * return values in cdw0. The latter is returned in the result pointer,
+	 * the former two generate return values here
+	 */
+
+	if (resp->hdr_len < sizeof(*resp_hdr)) {
+		errno = -EPROTO;
+		return -1;
+	}
+	resp_hdr = (struct nvme_mi_service_resp_hdr *)resp->hdr;
+
+	/* If we have a MI error, we can't be sure there's an admin header
+	 * following; return just the MI status, with the status type
+	 * indicator of MI.
+	 */
```

Comment:
This comment (and logic) is also incorrect, there is no admin header expected here.

Since we have checked that the size is valid to hold the entire response, there's no conditional parsing of the response required; just extract the status and cpsr values.

Reply:
yup, remove it